piccolo/apps/schema/commands/generate.py

```diff
@@ -251,28 +267,134 @@ async def get_table_schema(
 
 
 async def get_foreign_key_reference(
-    table_class: t.Type[Table], constraint_name: str
-) -> t.Optional[str]:
+    table_class: t.Type[Table], constraint_name: str, constraint_schema: str
+) -> ConstraintTable:
     """
     Retrieve the name of the table that a foreign key is referencing.
     """
     response = await table_class.raw(
         (
-            "SELECT table_name "
+            "SELECT table_name, table_schema "
             "FROM information_schema.constraint_column_usage "
-            "WHERE constraint_name = {};"
+            "WHERE constraint_name = {} AND constraint_schema  = {};"
         ),
         constraint_name,
+        constraint_schema,
     )
     if len(response) > 0:
-        return response[0]["table_name"]
+        return ConstraintTable(
+            name=response[0]["table_name"], schema=response[0]["table_schema"]
+        )
     else:
-        return None
+        return ConstraintTable()
+
+
+def get_table_name(name: str, schema: str) -> str:
+    if schema == "public":
+        return name
+    return f"{schema}.{name}"
 
 
-async def get_output_schema(schema_name: str = "public") -> OutputSchema:
+async def create_table(
+    Schema: t.Type[Table],
+    tablename: str,
+    schema_name: str,
+    output_schema: OutputSchema,
```

Comment:
Passing in an object an manipulating can sometimes be troublesome.

How about we don't pass in the `output_schema`, but this function returns a dataclass or tuple containing the table class, any warnings and any import strings. The calling function then updates their `output_schema` from the response.

Reply:
@dantownsend I didn't know it can be troublesome. but I can't really understand the reason. If it's not a trouble for you, can you please explain it to me a little bit?
I thought this should be pretty safe because we only use append operations on the lists and not actually manipulating the data inside of them.

Reply:
Yeah, maybe troublesome is the wrong word. It's more of a preference. It's easier to reason about a function if it's pure - and just returns stuff without any side effects.

I'm happy to keep it as it is.

Reply:
Ok now I understand your point and I agree with you. That's more natural. 
I'll change this.

Reply:
@dantownsend I'm sorry but I couldn't find a clean way to make this change. I just kept adding to the complexity. I would appreciate your help in this matter.

Reply:
@AliSayyah how about this - rather than `create_table` returning a table class, have it return an `OutputSchema`.

Then modify `OutputSchema` so it has a custom add method like this:

```python
class OutputSchema:
    ...
    def __add__(self, value) -> OutputSchema:
        self.imports.extend(value.imports)
        self.warnings.extend(value.warnings)
        self.tables.extend(value.tables)
```

So then you can just add the returned `OutputSchema` to the existing one, rather than passing it in as an argument and mutating it within the `create_table`.

With the part  where you're using `asyncio.gather` you can then do:

```python
output_schemas = await asyncio.gather(*tablecoroutines)

output_schema = sum(output_schemas, start=OutputSchema())
```

Does that make sense? It's the cleanest approach I can think of.

Reply:
That's a very good idea, thanks.

Reply:
The start parameter of the sum function has been added in python 3.8.

So this is my workaround:
```python
    def __radd__(self, value) -> OutputSchema:
        if isinstance(value, int):
            return self
        value.imports.extend(self.imports)
        value.warnings.extend(self.warnings)
        value.tables.extend(self.tables)
        return value

    def __add__(self, value) -> OutputSchema:
        self.imports.extend(value.imports)
        self.warnings.extend(value.warnings)
        self.tables.extend(value.tables)
        return self
```

Now I'm stuck with a bunch of lint errors:))

> piccolo/apps/schema/commands/generate.py:453: error: Item "int" of "Union[Any, int]" has no attribute "tables"
piccolo/apps/schema/commands/generate.py:454: error: Item "int" of "Union[Any, int]" has no attribute "tables"
piccolo/apps/schema/commands/generate.py:456: error: Item "int" of "Union[Any, int]" has no attribute "imports"
piccolo/apps/schema/commands/generate.py:460: error: Item "int" of "Union[Any, int]" has no attribute "imports"
piccolo/apps/schema/commands/generate.py:464: error: Incompatible return value type (got "Union[Any, int]", expected "OutputSchema")


is it ok to ignore these with `# type: ignore`?

Reply:
Yeah, you can ignore them if you like. Maybe add a type annotation to value, so it's `value: OutputSchema` helps?

It's a shame about the `start` parameter.

Reply:
unfortunately, It doesn't help.

Reply:
`type: ignore` is fine then